### PR TITLE
autocode, azure-cli, generate-json-schema, elixirscript, kibana: use Language::Node

### PR DESCRIPTION
**Pull request by chrmoritz**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

This PR is a flow up of https://github.com/Homebrew/brew/pull/37. It makes use of the `Language::Node` helper methods in all node module based formula (except [`tegh`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tegh.rb)), fixing npm cache issues and changing the module installation format from local to global style. (also fixing audit warnings for `kibana`)

In my opinion `tegh` should be moved to the boneyard, because it wasn't updated in the last 2.5 years, it's native addons part doesn't compile against any recent node version and even the tarball for brew (which does not contain any of the somewhat optional native addon dependencies) will work, but emits warnings that some features are not supported (most likely because of the missing native addons).

### CODE REVIEW

**Comment by DomT4** [comment]
> In my opinion tegh should be moved to the boneyard, because it wasn't updated in the last 2.5 years, it's native addons part doesn't compile against any recent node version and even the tarball for brew (which does not contain any of the somewhat optional native addon dependencies) will work, but emits warnings that some features are not supported (most likely because of the missing native addons).

Kill it.

**Comment by DomT4** [comment]
`kibana` failed on the first run, slightly oddly:

```
installing /usr/local/Cellar/kibana/4.5.0/libexec/node/include/node/zconf.h
installing /usr/local/Cellar/kibana/4.5.0/libexec/node/include/node/zlib.h
==> npm install --verbose
Failed to execute: npm
```

**Comment by DomT4** [comment]
Oh, I see, It looks like it was perhaps using a custom internal `npm`?
```ruby
ENV.prepend_path "PATH", prefix/"libexec/node/bin"
```

**Comment by chrmoritz** [comment]
The problem is that `kibana`was previously using the `npm` from it's bundled `node4` installation (which is compiled `--with-npm`). With the changes it tries to use the npm from the latest stable node installation (therefore it worked for me locally).

Possible way to fix this:
* add a `depends_on "node"` (which is otherwise quite redundant)
* extend `Language::Node` to support custom npm prefixes
* don't use `Language::Node` at all in `kibana` and set up the npm cache fix by hand in the formula

**Comment by DomT4** [comment]
In order of preference 3, 2 then 1. We'd probably want to exempt `kibana` from the `audit` code though in that situation.

**Comment by chrmoritz** [comment]
I've implemented 3 while still using `Language::Node` to grabbing our npm config (so that we can manage our `npm_cache` path in a single place).

> We'd probably want to exempt kibana from the audit code though in that situation.

Which wouldn't be new for `kibana`: https://github.com/Homebrew/brew/commit/bc98fd37882c64c896dc2243fcc6e129f170a32a#diff-fab7ad24732f89faa5deb3bb1f253a52L660

File: Formula/kibana.rb [hunk 1]
```diff
@@ -45,8 +47,10 @@ def install
     # do not build zip package
     inreplace buildpath/"tasks/build/archives.js", /(await exec\('zip'.*)/, "// \\1"
 
+    # set npm env and fix cache edge case (https://github.com/Homebrew/brew/pull/37#issuecomment-208840366)
     ENV.prepend_path "PATH", prefix/"libexec/node/bin"
-    system "npm", "install"
+    Pathname.new("#{ENV["HOME"]}/.npmrc").write Language::Node.npm_cache_config
```
Comment (DomT4) [inline Formula/kibana.rb]: 👍 to this but here you shouldn't need the `Pathname.new` element.

Comment (chrmoritz) [inline Formula/kibana.rb]: What would be the best ruby way to do this?
`"#{ENV["HOME"]}/.npmrc").write` and `ENV["HOME"]/".npmrc").write` do not work.

Comment (DomT4) [inline Formula/kibana.rb]: Oh, I thought the latter would work here but apparently not in this case. It's fine as-is then.

**Comment by chrmoritz** [comment]
The only remaining build failure is related to a audit error in the `kibana` formula, which was fixed in https://github.com/Homebrew/brew/pull/161 by white listing it.

**Comment by DomT4** [comment]
Merged! New audit code worked as expected, which is always nice. Thank you once more for getting this done @chrmoritz.

**Pull request closed without merging**